File: mario.py
from cs50 import get_int

# Get integer n from user
n = get_int("Provide a half-pyramid height between 1 and 8: ")

while n < 1 or n > 8:
    n = get_int("Provide a half-pyramid height between 1 and 8: ")

# Construction of pyramid
for i in range(n):

    # Make integer m (resiprocal to n)
    m = n-i-1

    # Print m-related spaces
    for j in range(m):
        print(" ", end="")

    # Print n-related #
    for k in range(i+1):
        print("#", end="")
        
    # Print middle spaces
    print("  ", end="")
    
    # Print n-related #
    for o in range(i+1):
        print("#", end="")
        
    # Print new line
    print("")

File: mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
// User input
{
    int n;
    do 
{
    n = get_int("Height: ");
}
while (n<1 || n>8);

for (int i = 0; i < n; i++)
{
// make integer m - resiprocal to n 
  int m = n -1 -i;
// print m-related spaces
   for (int j = 0; j < m ; j++)
   { 
            printf(" ");
   }
// print n-related #
   for (int j = 0; j <= i ; j++)
   { 
       printf("#");
   }
// print middle spaces
   printf("  ");
// print n-related #
    for (int j = 0; j <= i ; j++)
   { 
       printf("#");
   }
// print new line
   printf("\n");
}
}
